src/AmpWpPlugin.php

```diff
@@ -106,7 +110,20 @@ protected function get_bindings() {
 	 *                      to argument values.
 	 */
 	protected function get_arguments() {
-		return [];
+		return [
+			ServerTiming::class => [
+				// Wrapped in a closure so it is lazily evaluated. Otherwise,
+				// is_user_logged_in() breaks because it's used too early.
+				'verbose' => static function () {
+					return is_user_logged_in()
+						&& isset( $_GET[ QueryVar::VERBOSE_SERVER_TIMING ] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+						&& filter_var(
+							$_GET[ QueryVar::VERBOSE_SERVER_TIMING ], // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+							FILTER_VALIDATE_BOOLEAN
+						);
```

Comment:
Minor point, not relevant here, but `$_GET` is slashed. So in the general case, this should be:

```suggestion
						&& filter_var(
							wp_unslash( $_GET[ QueryVar::VERBOSE_SERVER_TIMING ] ), // phpcs:ignore WordPress.Security.NonceVerification.Recommended
							FILTER_VALIDATE_BOOLEAN
						);
```

Reply:
Would this be relevant for a supposedly boolean-ish value?

Reply:
No, not for a boolean string, but just for the general case where arbitrary strings are incoming.